src/App/Fossa/VSI/IAT/Resolve.hs

```diff
@@ -39,18 +41,51 @@ resolveRevision apiOpts locator =
     -- Since users may only register revision assertions as a byproduct of running CLI analysis,
     -- revision assertions are always for "custom" projects.
     if VSI.isTopLevelProject locator
-      then Fossa.resolveProjectDependencies apiOpts locator
-      else pure []
+      then recover $ Fossa.resolveProjectDependencies apiOpts locator
+      else pure . Just $ []
 
-resolveGraph :: (Has (Lift IO) sig m, Has Diagnostics sig m) => ApiOpts -> [VSI.Locator] -> m (Graphing VSI.Locator)
-resolveGraph apiOpts locators = context ("Resolving graph for " <> toText (show $ fmap VSI.renderLocator locators)) $ do
-  subgraphs <- traverse (resolveSubgraph apiOpts) locators
-  pure $ mconcat subgraphs
+resolveGraph :: (Has (Lift IO) sig m, Has Diagnostics sig m) => ApiOpts -> [VSI.Locator] -> VSI.SkipResolution -> m (Graphing VSI.Locator)
```

Comment:
The main reason this isn't tested (along with the rest of VSI/IAT stuff) is that I'm not aware of a good way to test this section of the codebase since it relies almost entirely on external stuff (a CLI plugin, remote services).

I'd love to get your thoughts on how we could better test this within our testing framework but yeah, I think for now I'll skip, since I think adding tests to VSI/IAT is more of a refactor than a simple thing to do.